### PR TITLE
Change Jetpack deployments to DVR by default

### PR DESCRIPTION
**Pull request by rajeshP524**
This PR resolves issue https://github.com/redhat-performance/jetpack/issues/438 and https://github.com/redhat-performance/jetpack/issues/496

### CODE REVIEW

File: group_vars/all.yml [hunk 1]
```diff
@@ -280,3 +280,8 @@ manila_enabled: false
 
 #image path for overcloud nodes
 remote_image_path: https://sf.redhat.com
+
+#DVR is the default for ML2/OVN deployments for osp releases >= 16
+#change below variable to false to disable DVR
+#Note: if Neutron_backend == 'ovs' or osp_release < 16, then DVR won't be enabled irrespective of setting below variable to true/false
```
Comment (sanjaychari) [inline group_vars/all.yml]: Do not have this feature. DVR is supported for ML2/OVS also, and is also supported in OSP 13. So change line 335 in overcloud.yml to only check for dvr_enabled == false.
Also change line 344 in overcloud.yml similarly.

**Review by sanjaychari — CHANGES_REQUESTED** [review]

File: group_vars/all.yml (hunk 1 above)
Comment (rajeshP524) [reply group_vars/all.yml]: > Do not have this feature. DVR is supported for ML2/OVS also, and is also supported in OSP 13. So change line 335 in overcloud.yml to only check for dvr_enabled == false. Also change line 344 in overcloud.yml similarly.

ack

**Comment by pradiptapks** [comment]
@rajeshP524 @sanjaychari @asyedham 

Please find my below suggestion:

- There are a couple of additional changes needed in DVR mapping via Jetpack.
- The compute interface template should rename br-ex to another name "br-dvr". 
- Advantage: When the External (ingress/egress) traffic is processed, the traffic should not be broadcast to the controller br-ex bridge.
- To do this Jetpack has to replace br-ex bridge to br-dvr in compute the interface template
```
-
  type: ovs_bridge
  name: br-dvr
  use_dhcp: false
  members:
    -
      type: interface
      name: ens7f1
      primary: true
```

- The network environment should have a correct mapping to the bridge.
```
  NeutronNetworkVLANRanges:
    - dvr:410:410
  NeutronBridgeMappings:
    - dvr:br-dvr
```
- A VLAN gateway route should be created in Undercloud for DVR network reachability. Example:
```
@undercloud ~]$ ip addr show ens7f1.410
12: ens7f1.410@ens7f1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 3c:fd:fe:c5:01:11 brd ff:ff:ff:ff:ff:ff
    inet 192.168.4.1/24 scope global ens7f1.410
       valid_lft forever preferred_lft forever

```

File: group_vars/all.yml (hunk 1 above)
Comment (pradiptapks) [reply group_vars/all.yml]: @sanjaychari @rajeshP524 

Before any integration, you can check Red Hat OpenStack Platform Life Cycle document for OSP release support

As per the data, OSP13 has maintenance support already expired. I don't think we need any major scale or performance test plan to execute in OSP13 release cycle. So we can focus on OSP16.x and 17.x release.


https://access.redhat.com/support/policy/updates/openstack/platform

**Comment by sanjaychari** [comment]
@pradiptapks 

I had a query. As far as I understand, for east-west traffic and north-south traffic with floating IPs in IPv4; DVR will only route traffic through the compute nodes and will not broadcast traffic to br-ex on the controller node, irrespective of whether br-ex or another bridge is used on the compute node. For north-south traffic without floating IP, traffic has to travel through the controller node anyway.
In this case, what would be the advantage of having a separate br-dvr bridge on compute nodes?

**Comment by rajeshP524** [comment]
> @rajeshP524 @sanjaychari @asyedham
> 
> Please find my below suggestion:
> 
> * There are a couple of additional changes needed in DVR mapping via Jetpack.
> * The compute interface template should rename br-ex to another name "br-dvr".
> * Advantage: When the External (ingress/egress) traffic is processed, the traffic should not be broadcast to the controller br-ex bridge.
> * To do this Jetpack has to replace br-ex bridge to br-dvr in compute the interface template
> 
> ```
> -
>   type: ovs_bridge
>   name: br-dvr
>   use_dhcp: false
>   members:
>     -
>       type: interface
>       name: ens7f1
>       primary: true
> ```
> 
> * The network environment should have a correct mapping to the bridge.
> 
> ```
>   NeutronNetworkVLANRanges:
>     - dvr:410:410
>   NeutronBridgeMappings:
>     - dvr:br-dvr
> ```
> 
> * A VLAN gateway route should be created in Undercloud for DVR network reachability. Example:
> 
> ```
> @undercloud ~]$ ip addr show ens7f1.410
> 12: ens7f1.410@ens7f1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
>     link/ether 3c:fd:fe:c5:01:11 brd ff:ff:ff:ff:ff:ff
>     inet 192.168.4.1/24 scope global ens7f1.410
>        valid_lft forever preferred_lft forever
> ```

Thanks @pradiptapks for these suggestions.
Leaving the name as "br-ex" on compute nic template as "br-ex" is removed from controller.

File: group_vars/all.yml [hunk 2]
```diff
@@ -207,7 +207,7 @@ vm_image_url:
    rhel8: http://download.hosts.prod.upshift.rdu2.redhat.com/released/RHEL-8/8.1.0/BaseOS/x86_64/images/rhel-guest-image-8.1-263.x86_64.qcow2
    rhel8_2: http://download.eng.pek2.redhat.com/released/RHEL-8/8.2.0/BaseOS/x86_64/images/rhel-guest-image-8.2-290.x86_64.qcow2
 #Neutron backend, if not set default will be used.
-#neutron_backend: ovn
+neutron_backend: ovn
```
Comment (sanjaychari) [inline group_vars/all.yml:213]: Add a comment that we are not supporting ml2/ovs for DVR right now because it needs changes in roles_data.yaml file.

**Review by sanjaychari — CHANGES_REQUESTED** [review]
Remove the swp file also.

File: group_vars/all.yml (hunk 2 above)
Comment (rajeshP524) [reply group_vars/all.yml:213]: done, added!

**Comment by rajeshP524** [comment]
> Remove the swp file also.

done

**Review by sanjaychari — APPROVED** [review]

**Review by pradiptapks — APPROVED** [review]
Regarding the control-plane network in `br-ex`, I had a syncup with Rajesh and Sanjay, and the mapping is based on the condition basis on which the node has 3 interfaces.

File: templates/network-environment_v2.yaml.j2 [hunk 3]
```diff
@@ -27,10 +27,17 @@ parameter_defaults:
...
       NeutronBridgeMappings: "datacentre:br-ex,tenant:br-isolated"
 {% if composable_roles %}
+{% if dvr_enabled == false %}
 {% for node_type in machine_types %}
     Compute{{ node_type }}Parameters:
       NeutronBridgeMappings: ""
 {% endfor %}
+{% endif %}
 {% endif %}
     NeutronExternalNetworkBridge: ""
```
Comment (asyedham) [inline templates/network-environment_v2.yaml.j2:37]: for dvr NeutronExternalNetworkBridge: "br-ex"?

**Pull request merged**